core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -2209,6 +2209,14 @@ private void processTxFromBestChain(Transaction tx, boolean forceAddToPool) thro
         if (!doubleSpendTxns.isEmpty()) {
             // no need to addTransactionsDependingOn(doubleSpendTxns) because killTxns() already kills dependencies;
             killTxns(doubleSpendTxns, tx);
+
+            // disconnect irrelevant inputs (otherwise might cause serialization issue)
+            for(TransactionInput input : tx.getInputs()) {
```

Comment:
Just one more thought about this first fix: what if the double spending transaction *is* relevant to the wallet (either directly by the double-spent UTXO or indirectly somehow). Is it still ok to disconnect the inputs? Do we already have a test for that?